Modules/Sources/WordPressUI/Views/Users/ViewModel/UserChangePasswordViewModel.swift

```diff
@@ -0,0 +1,76 @@
+import SwiftUI
+
+public class UserChangePasswordViewModel: ObservableObject {
+
+    public enum Errors: LocalizedError {
+        case passwordMustNotBeEmpty
+
+        public var errorDescription: String? {
+            switch self {
+                case .passwordMustNotBeEmpty: NSLocalizedString(
+                    "userchangepassword.error.empty",
+                    value: "Password must not be empty",
+                    comment: "An error message that appears when an empty password has been entered"
+                )
+            }
+        }
+    }
+
+    @Published
+    var password: String = ""
+
+    @Published
+    var isChangingPassword: Bool = false
+
+    @Published
+    var error: Error? = nil
+
+    @Environment(\.dismiss)
+    var dismissAction
+
+    private let actionDispatcher: UserManagementActionDispatcher
+    let user: DisplayUser
+
+    init(user: DisplayUser, actionDispatcher: UserManagementActionDispatcher) {
+        self.user = user
+        self.actionDispatcher = actionDispatcher
+    }
+
+    func didTapChangePassword(callback: @escaping () -> Void) {
+
+        if password.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty {
+            withAnimation {
+                error = Errors.passwordMustNotBeEmpty
+            }
+            return
+        }
+
+        withAnimation {
+            error = nil
+        }
+
+        Task {
+            await MainActor.run {
```

Comment:
You typically don't want to use `await MainActor.run` imperatively. The better option is to declare that `Task` runs is isolated to the main actor:

```swift
Task { @MainActor in
   // remove await MainActor.run
}
```

I'd also consider isolating the entire ViewModel to the main actor to ensure none of its properties are updated in the background.